packages/client-direct/src/index.ts

```diff
@@ -222,20 +222,46 @@ export class DirectClient {
 
                 await runtime.evaluate(memory, state);
 
-                const _result = await runtime.processActions(
-                    memory,
-                    [responseMessage],
-                    state,
-                    async (newMessages) => {
-                        message = newMessages;
-                        return [memory];
-                    }
+                // Check if we should suppress the initial message
+                const action = runtime.actions.find(
+                    (a) => a.name === response.action
                 );
+                const shouldSuppressInitialMessage =
+                    action?.suppressInitialMessage;
+
+                if (!shouldSuppressInitialMessage) {
+                    const _result = await runtime.processActions(
```

Comment:
this is basically now the same code (just the processActions instruction not the res.json) on both sides of the branch, should we hoist this up above the if or did I read this wrong?

Reply:
You are correct, I've gone ahead and fixed this.